### PR TITLE
#1114: changes for issue 

### LINKED ISSUE #1114: Inconsistent `SETBIT`: Handling negative and large offset values
## Steps to reproduce

1. Start the dicedb server and connect to it through whatever client(redis-cli or something else) you prefer.
2. Run either of the following commands
    a. `SETBIT <key> -1 1` without setting the `<key>` to any value.
    b. Set a value corresponding to a key, and then run `SETBIT <key> <large_negative_number> 1`. Make sure the `abs(<large_negative_number>)` is greater than the bit length of the value corresponding to the key.
3. Both will crash the dicedb server, and will close the connection.

## Expected output

This should throw an out of range error for negative offsets.
```
(error) ERR bit offset is not an integer or out of range
```

## Observed output

The observed output when the above set of commands when run on DiceDB

Same for both the cases.
```
Error: Server closed the connection
```

## Expectations for resolution

This issue will be considered resolved when the following things are done

1. Changes in the [`dice`](https://github.com/dicedb/dice) code to meet the expected behavior
2. Addition of relevant test case to ensure we catch the regression

You can find the tests under the `integration_tests` directory of the [`dice`](https://github.com/dicedb/dice) repository and the steps to run are in the [README file](https://github.com/dicedb/dice). Refer to the following links to set up DiceDB and Redis 7.2.5 locally

- [setup DiceDB locally](https://github.com/dicedb/dice)
- [setup Redis 7.2.5 locally](https://gist.github.com/arpitbbhayani/94aedf279349303ed7394197976b6843)
- [setup DiceDB CLI](https://github.com/dicedb/dice)

## Follow up
Once these changes are done, drop a comment in the issue #813 regarding the closing of this issue.

### PR DESCRIPTION
**Pull request by Yashasv-Prajapati**
This PR closes #1114.

### CODE REVIEW

**Comment by Yashasv-Prajapati** [comment]
Hey @apoorvyadav1111, could you please take a look at it and approve.

File: integration_tests/commands/async/bit_test.go
Comment (apoorvyadav1111) [inline integration_tests/commands/async/bit_test.go:1]: Please add tests in resp, websocket, and HTTP folder. We are migrating away from asyncserver.

**Review by apoorvyadav1111 — CHANGES_REQUESTED** [review]
Thanks for this fix, have left some comments.

File: integration_tests/commands/async/bit_test.go
Comment (Yashasv-Prajapati) [reply integration_tests/commands/async/bit_test.go:1]: hey, done as asked, please take a look and let me know

**Review by apoorvyadav1111 — APPROVED** [review]
Thanks for adding tests. will merge once checks are ✅

**Pull request merged**